Mathlib/Algebra/BigOperators/Group/List/Defs.lean

```diff
@@ -104,3 +105,17 @@ theorem prod_hom_rel (l : List ι) {r : M → N → Prop} {f : ι → M} {g : ι
 end Monoid
 
 end List
+
+-- This can also be moved to `Algebra/BigOperators/Group/Vector/Defs.lean`.
```

Comment:
Please do so

Reply:
Created this new file and updated. However, I reviewed this lemma again and see the following updates:

1. It's similar to those suggested to be removed in https://github.com/leanprover-community/mathlib4/pull/29400#discussion_r2566729800.
1. `prod_cons` for `List` has been moved to batteries in https://github.com/leanprover-community/batteries/blame/f2aca6fc4a47c5b67fad08d3eda5e949d5b73ac0/Batteries/Data/List/Lemmas.lean#L985.
1. It's a single lemma in the newly created file `Algebra/BigOperators/Group/Vector/Defs.lean` and it's easily provable with `rfl`.

So if it's better just not to keep it, I can also just remove it.